app/collector/data_collection_engine.py

```diff
@@ -0,0 +1,193 @@
+        return self._process_object(constants.DATABASE_ROUTE, f_q_name, new_dict)
+
+    def _process_schema (self, schema_name, domain, layer, provider, database):
+    # Process the object schema.
+        f_q_name = [domain['name'], layer['name'], provider['name'], database['name'], schema_name]
+        f_q_name = self._format_fully_qualified_name(f_q_name)
+
+        new_dict = {
+          "name": schema_name,
+          "fully_qualified_name": f_q_name,
+          "display_name": schema_name,
+          "description": schema_name,
+          "domain_id": domain['id'],
+          "layer_id": layer['id'],
+          "database_id": database['id']
+        }
+
+        self.log.log_obj_collecting('schema', schema_name)
+        return self._process_object(constants.SCHEMA_ROUTE, f_q_name, new_dict)
+
+    def _process_table (self, generic_table : GenericTable, domain, layer, provider, database, schema):
+    # Process the object table.
+        f_q_name = [domain['name'], layer['name'], provider['name'], database['name'], schema['name'], generic_table.name]
+        f_q_name = self._format_fully_qualified_name(f_q_name)
+
+        # Assemble the columns of the table.
+        columns = []
+        for column_obj in generic_table.columns:
+            column_name = column_obj.name
+
+            column_type = str(column_obj.type).upper()
+            column_type = constants.SQLTYPES_DICT[column_type]
+
+            primary_key = column_obj.primary_key
+            nullable    = column_obj.nullable
+            unique      = column_obj.unique
+            is_metadata = 'metadata_' in column_name
+            columns.append(
+                {
+                  "name": column_name,
+                  "display_name": column_name,
+                  "description": column_name,
+                  "data_type": column_type,
+                  "primary_key": primary_key,
+                  "nullable": nullable,
+                  "unique": unique,
+                  "is_metadata": is_metadata
+                }
+            )
+
+        new_dict = {
+          "name": generic_table.name,
+          "fully_qualified_name":  f_q_name,
+          "display_name": generic_table.name,
+          "description": generic_table.name,
+          "type": "REGULAR",
+          "domain_id": domain['id'],
+          "layer_id": layer['id'],
+          "database_id":  database['id'],
+          "database_schema_id": schema['id'],
+          "columns": columns
+        }
+
+        self.log.log_obj_collecting('table', generic_table.name)
+        return self._process_object(constants.TABLE_ROUTE, f_q_name, new_dict)
+
+    def _get_providers(self, page):
+    # Load the database providers.
+        dict_param = {"page":page, "page_size":20}
+        _, result = get_request(constants.PROVIDER_ROUTE, None, dict_param=dict_param)
+        return result["items"], result["page"], result["page_count"]
+
+    def _execute_collection(self, provider):
```

Comment:
You have to look the DatabaseProviderIngestion. Users will be able to set filters using regex. Not all asset may be imported.